wandb/wandb_run.py

```diff
@@ -415,11 +415,26 @@ def save(self, id=None, program=None, summary_metrics=None, num_retries=None, ap
         return upsert_result
 
     def use_artifact(self, artifact, type=None, aliases=None):
+        """ Declare an artifact as an input to a run, call `download` or `file` on \
```

Comment:
I think we should talk about get_path() here. Being able to get remote references is an important use case. Maybe we need to revisit the get_path API if you don't like it.

Reply:
It's tempting to make a .file() and a .remote() for all files. But we don't want to provide a remote() if it was actually a reference logged by the user (if it's an internal artifact file, remote would raise), because we don't want people making reference artifacts to files that are internal artifact files. We can chat about this, what I just said is confusing.